### PR TITLE
jewel: build/ops: Add fix subcommand to ceph-disk, fix SELinux denials, and speed up upgrade from non-SELinux enabled ceph to an SELinux enabled one

### PR DESCRIPTION
**Pull request by b-ranto**
http://tracker.ceph.com/issues/19543
http://tracker.ceph.com/issues/20192
http://tracker.ceph.com/issues/20194

### CODE REVIEW

**Comment by ktdreyer** [comment]
Tracked via http://tracker.ceph.com/issues/19543

**Comment by b-ranto** [comment]
It is only the first commit that conflicts. I've updated the commit message to include the Conflicts.

**Comment by b-ranto** [comment]
Bumping, @smithfarm are you ok with the commit messages now?

**Comment by smithfarm** [comment]
@b-ranto Yes, thanks for fixing it.

**Comment by smithfarm** [comment]
This PR is properly staged and will undergo integration testing in due course.

**Comment by smithfarm** [comment]
@b-ranto If there are related (follow-on) fixes you want to backport, feel free to cherry-pick them into this PR. (Please add the related backport tracker URLs to the PR description if you do.)

**Comment by b-ranto** [comment]
I have pushed the cherry-picked commits for some follow on fixes:

#14871
#15490

**Comment by smithfarm** [comment]
OK, trackers and PR description updated

**Comment by smithfarm** [comment]
Jenkins re-test this please

File: ceph.spec.in [hunk 1]
```diff
@@ -552,6 +552,7 @@ Summary:	SELinux support for Ceph MON, OSD and MDS
 Group:		System Environment/Base
 Requires:	ceph-base = %{epoch}:%{version}-%{release}
 Requires:	policycoreutils, libselinux-utils
+Requires(post):	ceph-base = %{_epoch_prefix}%{version}-%{release}
```
Comment (smithfarm) [inline ceph.spec.in]: @b-ranto There is no `%_epoch_prefix` in this spec file. Please adjust.

**Comment by smithfarm** [comment]
Will also need #16208 once that PR gets merged into master.

**Comment by b-ranto** [comment]
I have fixed the _epoch_prefix bit and and backported PR #16208.

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by smithfarm** [comment]
This passed a ceph-disk suite at http://tracker.ceph.com/issues/20613#note-9 and an upgrade/client-upgrade suite at http://tracker.ceph.com/issues/20613#note-11

Still waiting on rados and upgrade/hammer-x.

**Comment by smithfarm** [comment]
This passed a rados suite (with 1 unrelated failure) and an upgrade/hammer-x suite (with 1 SELinux denial failure). In general, the rados and upgrade/hammer-x integration runs have some transient failures due to SELinux denials:

* http://pulpito.front.sepia.ceph.com/smithfarm-2017-08-23_08:57:27-upgrade:hammer-x-wip-jewel-backports-distro-basic-smithi/1554284/
* http://pulpito.front.sepia.ceph.com/smithfarm-2017-08-22_12:32:46-upgrade:hammer-x-wip-jewel-backports-distro-basic-smithi/1551593/
* http://pulpito.front.sepia.ceph.com/smithfarm-2017-08-21_19:38:42-rados-wip-jewel-backports-distro-basic-smithi/1547759/

@b-ranto Do you think these SELinux denials could be related to this PR?

**Comment by b-ranto** [comment]
@smithfarm These are all generated by ceph-mon command, not ceph-disk and all of them point to some mis-labelled log files. There have been a couple of fixes (mostly in packaging/spec file iirc) around that in master (even way back when it referred to kraken) and these likely did not get back-ported to jewel. At least not yet.

**Comment by smithfarm** [comment]
@tchaikov Pinging for review.

**Pull request merged**